pgdog/src/frontend/router/sharding/value.rs

```diff
@@ -157,3 +171,23 @@ impl<'a> Value<'a> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use uuid::Uuid;
+
+    #[test]
+    fn uuid_binary_parses_correctly() -> Result<(), Error> {
+        let raw_bytes: [u8; 16] = [0x11u8; 16];
+        let expected_uuid = Uuid::parse_str("11111111-1111-1111-1111-111111111111")?;
+
+        let value = Value {
+            data_type: DataType::Uuid,
+            data: Data::Binary(&raw_bytes),
+        };
+
+        assert_eq!(value.uuid()?, Some(expected_uuid));
+        Ok(())
+    }
+}
```

Comment:
is this testing anything or is this test theatre?

Reply:
Can't hurt!